src/providers/trusted_service/context/generate_random.rs

```diff
@@ -0,0 +1,18 @@
+// Copyright 2021 Contributors to the Parsec project.
+// SPDX-License-Identifier: Apache-2.0
+use super::error::Error;
+use super::ts_protobuf::{GenerateRandomIn, GenerateRandomOut};
+use super::Context;
+use log::info;
+use std::convert::TryInto;
+
+impl Context {
+    pub fn generate_random(&self, size: usize) -> Result<Vec<u8>, Error> {
+        info!("Handling GenerateRandom request");
+        let open_req: GenerateRandomIn = GenerateRandomIn {
+            size: size.try_into().unwrap(), // Can't fail usize->u64 conversion
```

Comment:
Couldn't it fail on 32 bits platforms? Looks like `impl From<std::num::TryFromIntError> for Error` is [there](https://github.com/parallaxsecond/parsec/blob/1b6a60f8e09ff15487dac0367e9fd656f60b8191/src/providers/trusted_service/context/error.rs#L61) so you can probably just use `?` instead of `unwrap` here?

Reply:
Not really, if `usize` is 32 bits then extending to `u64` can't fail. You can get an error the other way.

The implementation is [here](https://doc.rust-lang.org/src/core/convert/num.rs.html#351), and that macro is implemented [here](https://doc.rust-lang.org/src/core/convert/num.rs.html#172). For some reason they went with `TryFrom` even though there isn't any possible error.......

Reply:
Oh, but looking at that implementation, maybe `size as u64` would be better, like in the `try_from_unbounded` implementation linked above.

Reply:
Right that is the safe direction here. Could fail on 128 bits platforms 🤡 
I would personally prefer `try_into()?` if that works easily but I'm ok with the `as` if not. Just so that when reading the code you don't have to think about it too much

Reply:
Changed to `size: size.try_into()?` if that works